api/v1beta1/gcpmachine_types.go

```diff
@@ -209,6 +229,19 @@ type GCPMachineSpec struct {
 	// ShieldedInstanceConfig is the Shielded VM configuration for this machine
 	// +optional
 	ShieldedInstanceConfig *GCPShieldedInstanceConfig `json:"shieldedInstanceConfig,omitempty"`
+
+	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is "Migrate".
+	// +kubebuilder:validation:Enum=MIGRATE;TERMINATE;
```

Comment:
There's a disconnect between the comment and the enum here, though the enum is probably the one that should be changed.

Might be worth including in the comment what the accepted values are so that `kubectl explain` documents the valid values